src/main/java/com/algorand/algosdk/account/Account.java

```diff
@@ -386,6 +386,109 @@ public SignedTransaction appendMultisigTransaction(MultisigAddress from, SignedT
         }
     }
 
+    /**
+     * Sign LogicSig with account's secret key
+     * @param lsig LogicsigSignature to sign
+     * @return LogicsigSignature with updated signature
+     * @throws IOException
+     */
+    public LogicsigSignature signLogicsig(LogicsigSignature lsig) throws IOException {
+        Signature sig;
+        try {
+            byte[] bytesToSign = lsig.bytesToSign();
+            sig = this.rawSignBytes(bytesToSign);
+        } catch (NoSuchAlgorithmException ex) {
+            throw new IOException("could not sign transaction", ex);
+        }
+        lsig.sig = sig;
+        return lsig;
+    }
+
+    /**
+     * Sign LogicSig as multisig
+     * @param lsig LogicsigSignature to sign
+     * @param ma MultisigAddress to format multi signature from
+     * @return LogicsigSignature
+     * @throws IOException
+     */
+    public LogicsigSignature signLogicsig(LogicsigSignature lsig, MultisigAddress ma) throws IOException {
+        Ed25519PublicKey myPK = this.getEd25519PublicKey();
+        int myI = ma.publicKeys.indexOf(myPK);
+        if (myI == -1) {
+            throw new IllegalArgumentException("Multisig identity does not contain this secret key");
+        }
+        // now, create the multisignature
+        Signature sig;
+        try {
+            byte[] bytesToSign = lsig.bytesToSign();
+            sig = this.rawSignBytes(bytesToSign);
+        } catch (NoSuchAlgorithmException ex) {
+            throw new IOException("could not sign transaction", ex);
+        }
+
+        MultisigSignature mSig = new MultisigSignature(ma.version, ma.threshold);
+        for (int i = 0; i < ma.publicKeys.size(); i++) {
+            if (i == myI) {
+                mSig.subsigs.add(new MultisigSubsig(myPK, sig));
+            } else {
+                mSig.subsigs.add(new MultisigSubsig(ma.publicKeys.get(i)));
+            }
+        }
+        lsig.msig = mSig;
+        return lsig;
+    }
+
+    /**
+     * Appends a signature to multisig logic signed transaction
+     * @param lsig LogicsigSignature append to
+     * @return LogicsigSignature
+     * @throws IllegalArgumentException
+     * @throws NoSuchAlgorithmException
+     */
+    public LogicsigSignature appendToLogicsig(LogicsigSignature lsig) throws IllegalArgumentException, IOException {
+        Ed25519PublicKey myPK = this.getEd25519PublicKey();
+        int myI = -1;
+        for (int i = 0; i < lsig.msig.subsigs.size(); i++ ) {
+            MultisigSubsig subsig = lsig.msig.subsigs.get(i);
+            if (subsig.key.equals(myPK)) {
```

Comment:
Was the intent to compare references? if so, consider replacing it with `==`, it's the same and cleaner. But, I believe the code intended to compare the contents of the arrays. Use `java.util.Arrays.equals(subsig.key, myPK)` instead.

Reply:
didn't notice you overrides equals. ignore the comment